normandy/base/api/mixins.py

```diff
@@ -12,3 +15,32 @@ def retrieve(self, request, *args, **kwargs):
     @cache_control(public=True, max_age=settings.API_CACHE_TIME)
     def list(self, request, *args, **kwargs):
         return super().list(request, *args, **kwargs)
+
+
+_midair_collision_enabled = True
+
+
+def _enable_midair_collision(enabled=True):
+    global _midair_collision_enabled
+    _midair_collision_enabled = enabled
+
+
+class MidairCollisionViewsetMixin(object):
+    """Modify a ModelViewSet to add midair collision to update methods"""
+
+    def update(self, request, *args, **kwargs):
+        if _midair_collision_enabled:
+            # Using a serializer gets some easy validation
+            serializer = LastUpdatedSerializer()
+            request_ts = serializer.to_internal_value(request.data)['last_updated']
```

Comment:
Yes, though it ends up being slightly longer:

```
serializer = LastUpdatedSerializer(data=request.data)
serializer.is_valid(raise_exception=True)
request_ts = serializer.validated_data['last_updated']
```

The call to `is_valid` is required, without it, accessing `validated_data` throws an error (I wish it would just run the validation, instead of raising an error). This is probably more readable though, what do you think?